nautobot/core/utils/lookup.py

```diff
@@ -208,6 +213,20 @@ def get_view_for_model(model, view_type=""):
     return result
 
 
+def get_model_for_view_name(view_name):
+    """
+    Return the model class associated with the given view_name e.g. "circuits:circuit_detail", "dcim:device_list" and etc.
+    If the app_label or model_name contained by the given view_name is invalid, this will return `None`.
+    """
+    app_label, model_name = view_name.split(":")  # dcim, device_list
+    model_name = model_name.split("_")[0]  # device
```

Comment:
Does this need the reverse equivalent of the logic in `get_view_for_model`, e.g. `user -> users`, etc.?

Reply:
No longer needed after resolution in standup